test/xref2/subst/test.md

```diff
@@ -76,10 +76,141 @@ type vv/5 = local(SubstituteMe/2,false).v
+module type Monad/30 = sig
+  type t/31
+  val map/32 : ([a] t/31) -> ((a) -> b) -> [b] t/31
+  val join/33 : ([[a] t/31] t/31) -> [a] t/31
+   (removed=[])end
+module SomeMonad/29 : sig
+  type t/34
+  include : Monad/30 with [root(Monad/30).t = [a] t/34] (sig = val map/35 : ([a] t/34) -> ((a) -> b) -> [b] t/34
+                                                               val join/36 : ([[a] t/34] t/34) -> [a] t/34
+                                                                (removed=[]))
+   (removed=[])end
+module ComplexTypeExpr/27 : sig
+  type t/37
+  include : Monad/30 with [root(Monad/30).t = ([int * a] t/37 * [a * int] t/37)] (sig =
+    val map/38 : (([int * a] t/37 * [a * int] t/37)) -> ((a) -> b) -> ([int * b] t/37 * [b * int] t/37)
+    val join/39 : (([int * ([int * a] t/37 * [a * int] t/37)] t/37 * [([int * a] t/37 * [a * int] t/37) * int] t/37)) -> ([int * a] t/37 * [a * int] t/37)
+     (removed=[]))
+   (removed=[])end
+module Erase/28 : sig
+  include : Monad/30 with [root(Monad/30).t = a] (sig = val map/40 : (a) -> ((a) -> b) -> b
+                                                        val join/41 : (a) -> a
+                                                         (removed=[]))
+   (removed=[])end
+ (removed=[])
+```
+
+More tests with two type variables:
+
+```ocaml
+# compile {|
+  module type Monad_2 = sig
+    type ('a, 'err) t
+    val map : ('a, 'err) t -> f:('a -> 'b) -> ('b, 'err) t
+    val join : (('a, 'e) t, 'e) t -> ('a, 'e) t
+    val both : ('a, 'e) t -> ('b, 'e) t -> ('a * 'b, 'e) t
+  end
+
+  module SwappedVars : sig
+    type ('x, 'y) t
+    include Monad_2 with type ('a, 'b) t := ('b, 'a) t
+  end
+  |}
+- : Component.Signature.t =
+module type Monad_2/54 = sig
+  type t/55
+  val map/56 : ([a * err] t/55) -> f:((a) -> b) -> [b * err] t/55
+  val join/57 : ([[a * e] t/55 * e] t/55) -> [a * e] t/55
+  val both/58 : ([a * e] t/55) -> ([b * e] t/55) -> [(a * b) * e] t/55
+   (removed=[])end
+module SwappedVars/53 : sig
+  type t/59
+  include : Monad_2/54 with [root(Monad_2/54).t = [b * a] t/59] (sig =
+    val map/60 : ([err * a] t/59) -> f:((a) -> b) -> [err * b] t/59
+    val join/61 : ([e * [e * a] t/59] t/59) -> [e * a] t/59
+    val both/62 : ([e * a] t/59) -> ([e * b] t/59) -> [e * (a * b)] t/59
+     (removed=[]))
+   (removed=[])end
+ (removed=[])
+```
+
+Edge cases:
+
+```ocaml
+# compile {|
+  module type S = sig
+    type 'a t
+    val map : 'a t -> ('a -> 'b) -> 'b t
+  end
+
+  module M : sig
+    type 'a t
+    include S with type 'a t := ([ `A of 'a * 'b ] as 'b) t
+  end
+  |}
+- : Component.Signature.t =
+module type S/69 = sig
+  type t/70
+  val map/71 : ([a] t/70) -> ((a) -> b) -> [b] t/70
+   (removed=[])end
+module M/68 : sig
+  type t/72
+  include : S/69 with [root(S/69).t = [(alias (poly_var [ `A of (a * b) ]) b)] t/72] (sig =
+    val map/73 : ([(alias (poly_var [ `A of (a * b) ]) b)] t/72) -> ((a) -> b) -> [(alias (poly_var [ `A of (b * b) ]) b)] t/72
```

Comment:
I've found an edge case, the last type is wrong: `(alias (poly_var [ `A of (b * b) ]) b)`.
We should either rename `b` to `c` in this type or choose a free variable before substituting.
What do you think ?